### ISSUE TITLE
Incorrect documentation around templating

### ISSUE THREAD

**Issue by MarqueIV** [gstroup/apimocker#84]
Hey!  Really great tool.  We depend on this heavily here at Citi for our mobile apps.

Just wanted to call out an issue we had that unfortunately was caused by incorrect documentation here in your 'readme.md' file.  Specifically, it's around templating.

In your example, you stated that for this configuration... 
```
 "template/:Name/:Number" :{
   "mockFile": "templateSample.json",
   "verbs":["get"],
   "enableTemplate": true
   "contentType":"application/json"
 }
```
With this template...
```
{
  "Name": "@Name",
  "Number": "@Number"
}
```
When you call `/John/12345` you claim it will return this:
```
{
	"Name": "John"
	"Number": 12345
}
```
But we're actually getting this...
```
{
	"Name": "John"
	"Number": "12345"  <-- Note the quotes
}
```
For one of our service calls, we needed the non-quote-wrapped number.

The confusion came in when we opened up our template in a JSON editor and tried removing the quotes around '@number' but it started barking at us that we now had invalid JSON, so thinking that wasn't the correct approach, we abandoned it.

The problem is that actually ***is*** the correct way to format the template to have it return the placeholder without quotes.  It just means that your template pre-filled-in is not valid JSON, and thus has to be edited with a text file.  That's what tripped us up.

If you put a note/caveat/warning/whatever stating that if you need the non-quoted version, your template will no longer be valid JSON, but _**is still valid for Mocker**_ (since it will become valid once merged) it would've gone a long way to helping people like us avoiding going down a rabbit hole trying to debug something we didn't actually have to debug.

That said, have you considered doing something like this for the tokens you don't want to have to wrap?
```
"#Number" or
"@@Number" or
"@Number@"
etc...
```
By implementing something like that (I'm personally partial to the '@@' myself) we can still keep our templates editable with a JSON editor without sacrificing output which is essential considering we run validation scripts against our JSON response files and now we have to explicitly exclude one. 

It shouldn't be too difficult to implement with RegEx either. Just search for that version first, consuming the quotes as part of the search, then run the parsing again with the second flavor just as you do now. (You may even be able to format a single RegEx that handles both!)

Anyway, hope this helps.  Again, keep up the really, really great work!

Mark

**Comment by ferrerod** [comment]
Quick solution is to implement the fix and issue a PR against the project with tests to show it works. I like your idea and so I will be issuing a PR with your proposal as well as some README fixes. Stay tuned.

**Comment by MarqueIV** [comment]
Yeah, I know about creating PR’s, but for some reason it won’t let me push, so I filed the report. I don’t know why our corporate network blocks github pushes but they do. 

Anyway, thanks for addressing this. I’m sure it will help someone.

[Sent from my... wait... does anyone really read this part?!?]

**Comment by gstroup** [comment]
Thanks for pointing this out.  And I’m glad to hear someone is using this thing!  😄 Is it a new problem?  Was this ever working?  Thanks.

**Comment by ferrerod** [comment]
Greg, Mark points out two main issues which I'll try to summarize:

- README example is incorrect with regard to the JSON returned from templated "@Number". Quotes do appear in returned JSON. 
- Using @Number without quotes in a JSON file means the template mock file is invalid JSON

**Comment by MarqueIV** [comment]
Hi guys!  Shared the update with the team here at Citi. Lots of happy people including management!

Quick thought... originally I was thinking @@tag was the best solution, but I e sort of convinced myself that @#tag is clearer. That keeps the single @ representing a replacement, and clearly shows you want a number.

What do you guys think?

**Comment by MarqueIV** [comment]
Crap! Damn iPhone!  Didn’t mean to close that. Sorry. Disregard.

**Comment by MarqueIV** [comment]
Hey David, saw you updated the PR.  I like the changes!  Question though... did you see my comment above about using "@#tag" instead of "@@tag"?  I think it reads more clear and separates intent.  Now '@' simply means 'a replacement tag' whereas '#' means '...with the number format rules'

**Comment by ferrerod** [comment]
I could go either way. My only concern with using @#Number instead of @@Numbe is just another syntax  to remember. @gstroup, what are your thoughts?

**Comment by gstroup** [comment]
Probably either syntax is OK.  @# or @@.  The only question I have is - what about a boolean value?  If we use something like "@@isActive", then it would return an actual boolean value, right?  Not the string "true"?  For this reason, I think the @@ syntax might be better.  @@ could apply to numbers, booleans, etc.  What do you guys think?

**Comment by MarqueIV** [comment]
@gstroup, good point about the boolean. I hadn't considered that, but you are right, booleans aren't numbers but don't appear in quotes either. 

Even more to the point, that's technically a json rule, not a 'response' rule.  What about xml responses? HTML? CSV?  I keep forgetting this tool is for mocking *any* APIs from a server, not just microservices returning json as we're using it.

Ok, so now the rule is simply: '@' means 'replace with value' and '@@' means 'replace with value while also consuming the quotes'.  The format of the value is irrelevant to templating.

Consider my mind changed.  Let's stick with @@ as it is now.
